src/pudl/transform/eia860.py

```diff
@@ -1157,3 +1157,172 @@ def cooling_equipment_continuity(cooling_equipment):
         groupby_col="report_date",
         n_outliers_allowed=1,
     )
+
+
+@asset(io_manager_key="pudl_io_manager")
+def _core_eia860__fgd_equipment(
+    raw_eia860__fgd_equipment: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform the EIA 860 FGD equipment table.
+
+    Transformations include:
+    - convert string booleans to boolean dtypes, and mixed strings/numbers to numbers
+    - convert kilodollars to normal dollars
+    - handle mixed reporting of percentages
+    - spot fix a duplicated SO2 control ID
+    - change an old water code to preserve detail of reporting over time
+    - add manufacturer name based on the code reported
+
+    """
+    fgd_df = raw_eia860__fgd_equipment
+
+    # Generic cleaning
+    fgd_df = fgd_df.pipe(pudl.helpers.fix_eia_na).pipe(pudl.helpers.add_fips_ids)
+
+    # Spot cleaning and date conversion
+    fgd_df = fgd_df.pipe(pudl.helpers.month_year_to_date).pipe(
+        pudl.helpers.convert_to_date
+    )
+    fgd_df = fgd_df.rename(columns={"operating_date": "fgd_operating_date"})
+
+    # Handle mixed boolean types in control flag column
+    for col in [
+        "byproduct_recovery",
+        "flue_gas_bypass_fgd",
+        "sludge_pond",
+        "sludge_pond_lined",
+    ]:
+        fgd_df = pudl.helpers.convert_col_to_bool(
+            df=fgd_df,
+            col_name=col,
+            true_values=["Y", "y"],
+            false_values=["N", "n"],
+        )
+
+    # Convert thousands of dollars to dollars and remove suffix from column name
+    fgd_df.loc[:, fgd_df.columns.str.endswith("_thousand_dollars")] *= 1000
+    fgd_df.columns = fgd_df.columns.str.replace("_thousand_dollars", "")
+
+    # Deal with mixed 0-1 and 0-100 percentage reporting
+    pct_cols = [
+        "flue_gas_entering_fgd_pct_of_total",
+        "so2_removal_efficiency_design",
+        "specifications_of_coal_ash",
+        "specifications_of_coal_sulfur",
+    ]
+    fgd_df = pudl.helpers.standardize_percentages_ratio(
+        frac_df=fgd_df, mixed_cols=pct_cols
+    )
+
+    # Fix duplicated SO2 control ID for plant 6016 in 2011
+    # Every other year 2009-2012 "01" refers to 2009 operating FGD equipment, and "1"
+    # refers to the 1976 operating FGD equipment. In 2011 the plant reports two "1"
+    # plants, so we change the SO2 control ID to "01" for the 2009 unit to preserve
+    # uniqueness.
+    fgd_df.loc[
+        (fgd_df.plant_id_eia == 6016)
+        & (fgd_df.report_date == "2011-01-01")
+        & (fgd_df.fgd_operating_date == "2009-03-01"),
+        "so2_control_id_eia",
+    ] = "01"
+
+    # Add a manufacturer name from the code.
+    fgd_df["fgd_manufacturer"] = fgd_df.fgd_manufacturer_code.map(
```

Comment:
Hm, should I have done this for any of the code columns in the cooling equipment? Seems slightly denormalizey but also probably useful to have the manufacturer name in here so 🤷

Reply:
This column doesn't seem to exist in the 860 cooling tables, unless I'm missing something?